coordinator/src/main/java/org/apache/uniffle/coordinator/conf/DynamicClientConfService.java

```diff
@@ -0,0 +1,171 @@
+        break;
+      case LEGACY:
+        confParsers = new ClientConfParser[] {new LegacyClientConfParser()};
+        break;
+      case MIXED:
+      default:
+        confParsers =
+            new ClientConfParser[] {new LegacyClientConfParser(), new YamlClientConfParser()};
+        break;
+    }
+    this.parsers = confParsers;
+
+    if (!fileSystem.isFile(confStoredPath)) {
+      String msg = String.format("Fail to init, %s is not a file.", confStoredPath.toUri());
+      LOGGER.error(msg);
+      throw new IllegalStateException(msg);
+    }
+    refreshClientConf();
+    LOGGER.info("Load client conf from {} successfully", confStoredPath);
+
+    int updateIntervalSec =
+        coordinatorConf.getInteger(
+            CoordinatorConf.COORDINATOR_DYNAMIC_CLIENT_CONF_UPDATE_INTERVAL_SEC);
+    updateClientConfExecutor =
+        ThreadUtils.getDaemonSingleThreadScheduledExecutor(this.getClass().getSimpleName());
+    updateClientConfExecutor.scheduleAtFixedRate(
+        this::refreshClientConf, 0, updateIntervalSec, TimeUnit.SECONDS);
+  }
+
+  private void refreshClientConf() {
+    try {
+      FileStatus[] fileStatus = fileSystem.listStatus(confStoredPath);
+      if (!ArrayUtils.isEmpty(fileStatus)) {
+        long modifiedMS = fileStatus[0].getModificationTime();
+        if (latestModificationMS.get() != modifiedMS) {
+          doRefreshClientConf();
+          latestModificationMS.set(modifiedMS);
+          Arrays.stream(callbacks).forEach(x -> x.accept(clientConf));
+          LOGGER.info("Update client conf from {} successfully.", confStoredPath);
+        }
+      } else {
+        LOGGER.warn("Client conf file not found with {}", confStoredPath);
+      }
+    } catch (Exception e) {
+      LOGGER.warn("Error when update client conf with {}.", confStoredPath, e);
+    }
+  }
+
+  private void doRefreshClientConf() throws Exception {
+    try (FSDataInputStream in = fileSystem.open(confStoredPath)) {
+      for (ClientConfParser parser : parsers) {
+        try {
+          ClientConf conf = parser.tryParse(in);
+          synchronized (clientConfLock) {
+            this.clientConf = conf;
+          }
+          return;
+        } catch (Exception e) {
+          // ignore
+        }
+      }
+    } catch (IOException e) {
+      LOGGER.error("Fail to refresh client conf from {}", confStoredPath.toUri().toString());
+      return;
+    }
+
+    throw new Exception("Unknown format of clientConf file.");
+  }
+
+  public Map<String, String> getRssClientConf() {
+    synchronized (clientConfLock) {
+      return clientConf.getRssClientConf();
+    }
+  }
+
+  public Map<String, RemoteStorageInfo> listRemoteStorageInfos() {
+    synchronized (clientConfLock) {
+      return clientConf.getRemoteStorageInfos();
+    }
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (updateClientConfExecutor != null) {
+      updateClientConfExecutor.shutdownNow();
```

Comment:
There's no need for a graceful shutdown, right?

Reply:
Yes. This part follows the original logic like `ClientConfManager` does